--- npm-3.10.8/lib/build.js.orig	2016-07-01 22:52:19.000000000 +0200
+++ npm-3.10.8/lib/build.js	2016-09-12 09:38:31.069484189 +0200
@@ -16,7 +16,7 @@
 var readJson = require('read-package-json')
 var link = require('./utils/link.js')
 var linkIfExists = link.ifExists
-var cmdShim = require('cmd-shim')
+var cmdShim = (process.platform === "win32") ? require("cmd-shim") : {}
 var cmdShimIfExists = cmdShim.ifExists
 var asyncMap = require('slide').asyncMap
 var ini = require('ini')
--- npm-3.10.8/package.json.orig	2016-09-09 03:21:47.000000000 +0200
+++ npm-3.10.8/package.json	2016-09-12 09:38:42.202817644 +0200
@@ -37,7 +37,6 @@
     "archy": "~1.0.0",
     "asap": "~2.0.4",
     "chownr": "~1.0.1",
-    "cmd-shim": "~2.0.2",
     "columnify": "~1.5.4",
     "config-chain": "~1.1.10",
     "dezalgo": "~1.0.3",
