--- npm-1.2.17/lib/build.js~	2013-04-03 19:11:45.000000000 +0300
+++ npm-1.2.17/lib/build.js	2013-04-04 22:37:51.277211534 +0300
@@ -16,7 +16,7 @@
   , readJson = require("read-package-json")
   , link = require("./utils/link.js")
   , linkIfExists = link.ifExists
-  , cmdShim = require("cmd-shim")
+  , cmdShim = (process.platform === "win32") ? require("cmd-shim") : {}
   , cmdShimIfExists = cmdShim.ifExists
   , asyncMap = require("slide").asyncMap
 
--- npm-1.4.15/package.json~	2014-06-11 04:31:44.000000000 +0300
+++ npm-1.4.15/package.json	2016-01-09 04:21:09.785619286 +0200
@@ -41,7 +41,6 @@
     "child-process-close": "~0.1.1",
     "chmodr": "~0.1.0",
     "chownr": "0",
-    "cmd-shim": "~1.1.1",
     "columnify": "~1.1.0",
     "editor": "~0.1.0",
     "fstream": "~0.1.26",
