--- npm-3.10.8/lib/link.js.orig	2016-07-01 22:52:19.000000000 +0200
+++ npm-3.10.8/lib/link.js	2016-09-12 09:55:21.476161853 +0200
@@ -8,6 +8,7 @@
 var asyncMap = require('slide').asyncMap
 var chain = require('slide').chain
 var path = require('path')
+var globalPaths = require('module').globalPaths
 var build = require('./build.js')
 var npa = require('npm-package-arg')
 var usage = require('./utils/usage')
@@ -98,11 +99,24 @@
       })
     }
 
-    fs.lstat(pp, function (er, st) {
-      if (er) {
+    // find pp from globalPaths
+    asyncMap(globalPaths, function(f, cb) {
+      f = path.resolve(f, pkg)
+      fs.lstat(f, function(er, st) {
+        if (er) {
+          cb()
+        } else {
+          cb(null, f, st)
+        }
+      });
+    }, function(er, f, st) {
+      pp = f && f[0] || path.resolve(npm.globalDir, pkg)
+
+      if (er) return cb()
+      if (!f || !f[0]) {
         rp = pp
         return npm.commands.install(t, [pkg], n)
-      } else if (!st.isSymbolicLink()) {
+      } else if (!st[0].isSymbolicLink()) {
         rp = pp
         next()
       } else {
